numba_cuda/numba/cuda/cudadrv/mappings.py

```diff
@@ -0,0 +1,23 @@
+from numba import config
+from cuda import cuda
+from . import enums
+if config.CUDA_USE_NVIDIA_BINDING:
+    jitty = cuda.CUjitInputType
+    FILE_EXTENSION_MAP = {
+        'o': jitty.CU_JIT_INPUT_OBJECT,
+        'ptx': jitty.CU_JIT_INPUT_PTX,
+        'a': jitty.CU_JIT_INPUT_LIBRARY,
+        'lib': jitty.CU_JIT_INPUT_LIBRARY,
+        'cubin': jitty.CU_JIT_INPUT_CUBIN,
+        'fatbin': jitty.CU_JIT_INPUT_FATBINARY,
```

Comment:
By my reading of [the docs](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__TYPES.html#group__CUDA__TYPES_1gc78e5cb421c428676861189048888958) I think this could be CU_JIT_INPUT_NVVM. Maybe `nvJitLinkInputType.NVJITLINK_INPUT_LTOIR` is the correct thing though? I am not sure this is exposed but if not I can see what might need to be done to expose it.

Reply:
You're right, it should be `CU_JIT_INPUT_NVVM` here - this mapping is using the enum from the driver API's `CUjitInputType`. However in `enums.py` the value for this is called `CU_JIT_INPUT_LTO_IR` in this PR at present. I don't mind which name we use for it as long as we're consistent.

 `nvJitLinkInputType.NVJITLINK_INPUT_LTOIR` is from a different enum, so even though it feels like it would be more correct for the nvjitlink case, we need to be consistent across all the linkers and use the driver enum values.